### PR TITLE
Re-implement clamp to use float min/max

### PR DESCRIPTION
**Pull request by mingmingtasd**
PTAL, thanks! @fujunwei

### CODE REVIEW

**Comment by mingmingtasd** [comment]
It will depend on https://github.com/webmachinelearning/webnn-samples/issues/88 and https://github.com/webmachinelearning/webnn-polyfill/issues/122

File: webnn.json [hunk 1]
```diff
@@ -216,8 +216,8 @@
   "clamp options": {
     "category": "structure",
     "members": [
-      {"name": "min value", "type": "operand", "optional": true},
-      {"name": "max value", "type": "operand", "optional": true}
+      {"name": "min value", "type": "float", "annotation": "*", "optional": true},
```
Comment (fujunwei) [inline webnn.json:219]: The json generator support complex expression like below:

`{"name": "min value", "type": "float", "default": "std::numeric_limits<float>::lowest()"},`

File: webnn.json [hunk 2]
```diff
@@ -216,8 +216,8 @@
...
+      {"name": "max value", "type": "float", "annotation": "*", "optional": true}
```
Comment (fujunwei) [inline webnn.json:220]: `{"name": "max value", "type": "float", "default": "std::numeric_limits<float>::max()"},`

**Review by fujunwei — COMMENTED** [review]
Well done, please take a look some comments .

File: src/webnn_native/dml/GraphDML.cpp [hunk 3]
```diff
@@ -510,11 +510,9 @@ namespace webnn_native { namespace dml {
                 break;
             }
             // Workaround(mingming): Currently we implement Relu6 operator by clamp. While
-            // Clamp operator is not supported for fusion, and to ensure that we can find
-            // the min and max operands from the graph, we have added a clamp node in
-            // GraphBuilder directly.
+            // Clamp operator is not supported for fusion, so we should add Clamp directly.
             case FusedOperator::Clamp:
-                break;
+                return dmlActivation;
```
[the remark this answers is not in the document]
Comment (mingmingtasd) [reply src/webnn_native/dml/GraphDML.cpp:508]: Refactored codes by CreateFusedActivation() and EmulateFusedActivation().

**Comment by mingmingtasd** [comment]
Updated, PTAL, thanks! @fujunwei

File: src/webnn_native/dml/GraphDML.cpp [hunk 4]
```diff
@@ -527,11 +518,36 @@ namespace webnn_native { namespace dml {
...
+                input = HardSwish(input);
+                return;
+            }
+            // Currently we implement Relu6 operator by clamp.
+            case FusedOperator::Clamp: {
+                auto clamp = reinterpret_cast<const op::Clamp*>(activation);
+                input = ::dml::Clip(input, clamp->GetMinValue(), clamp->GetMaxValue());
+                return;
+            }
+            case FusedOperator::Relu:
```
Comment (fujunwei) [inline src/webnn_native/dml/GraphDML.cpp:542]: CreateFusedActivation has check FusedOperator::Relu and FusedOperator::Sigmoid and FusedOperator::LeakyRelu, so it's better to use `if else` to check not fused op.

```
if (activation == nullptr) {
   return input;
}
auto type = activation->GetFusedOperator();
if (type == FusedOperator::HardSwish) {
   return HardSwish(input);
} else (type == FusedOperator::Clamp) {
   ....
}
return input;
```

**Review by fujunwei — APPROVED** [review]
LGTM with a open comment.

**Comment by BruceDai** [comment]
@mingmingtasd The CI failure of Node binding is due to clamp tests of webnn-polyfill being not updated to align latest spec changes https://github.com/webmachinelearning/webnn/pull/206. The pr https://github.com/webmachinelearning/webnn-polyfill/pull/123 has been submitted, PTAL, thanks.

File: src/webnn_native/dml/GraphDML.cpp (hunk 4 above)
Comment (mingmingtasd) [reply src/webnn_native/dml/GraphDML.cpp]: Yes, looks good, I updated, thanks! @fujunwei

**Comment by mingmingtasd** [comment]
> @mingmingtasd The CI failure of Node binding is due to clamp tests of webnn-polyfill being not updated to align latest spec changes [webmachinelearning/webnn#206](https://github.com/webmachinelearning/webnn/pull/206). The pr [webmachinelearning/webnn-polyfill#123](https://github.com/webmachinelearning/webnn-polyfill/pull/123) has been submitted, PTAL, thanks.

Thanks! @BruceDai

**Comment by mingmingtasd** [comment]
I have updated webnn-samples/webnn-polyfill commit id in DEPS, PTAL and help to merge, thanks! @fujunwei

**Comment by fujunwei** [comment]
Merge it.

**Pull request merged**